app/controllers/projects_controller.rb

```diff
@@ -516,10 +528,11 @@ def retrieve_projects
   # rubocop:enable Metrics/PerceivedComplexity
 
   def set_homepage_url
-    return if repo_data.nil?
+    retrieved_repo_data = repo_data
```

Comment:
I don't understand why you're assigning to a new variable here, since you don't seem to mutate it.

Reply:
Every call to "repo_data" retrieves a dataset from GitHub.  The old implementation downloaded the data from GitHub twice, once to see if it's nil and the other to actually search through it using "find".  This halves the number of requests to GitHub, so it's worth it!

Reply:
Good news: A few characters in Ruby can get a lot of work done.
Bad news: A few characters in Ruby can force the execution of a lot of work :-).